examples/test/test_bert_cuda_gpu.py

```diff
@@ -0,0 +1,37 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import os
+from pathlib import Path
+
+import pytest
+from utils import check_search_output, patch_config, update_azureml_config
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup():
+    """setup any state specific to the execution of the given module."""
+    cur_dir = Path(__file__).resolve().parent.parent
+    example_dir = cur_dir / "bert"
+    os.chdir(example_dir)
+    yield
+    os.chdir(cur_dir)
+
+
+@pytest.mark.parametrize("search_algorithm", ["tpe"])
+@pytest.mark.parametrize("execution_order", ["joint"])
+@pytest.mark.parametrize("system", ["aml_system"])
+@pytest.mark.parametrize("olive_json", ["bert_cuda_gpu.json"])
+@pytest.mark.parametrize("enable_cuda_graph", [True, False])
+def test_bert(search_algorithm, execution_order, system, olive_json, enable_cuda_graph):
+
+    from olive.workflows import run as olive_run
+
+    olive_config = patch_config(olive_json, search_algorithm, execution_order, system, is_gpu=True)
+    olive_config["passes"]["perf_tuning"]["config"]["enable_cuda_graph"] = enable_cuda_graph
+
+    update_azureml_config(olive_config)
+
+    footprint = olive_run(olive_config)
+    check_search_output(footprint)
```

Comment:
How do we verify if GPU was used in this test?

Reply:
The test will fail if CUDA ep is not found in onnxruntime available providers, or if no CUDA GPUs are available on the machine.